src/dmd/dtemplate.d

```diff
@@ -2543,6 +2546,7 @@ void functionResolve(ref MatchAccumulator m, Dsymbol dstart, Loc loc, Scope* sc,
             {
                 if (fd.fbody && !m.lastf.fbody) goto LfIsBetter;
                 if (!fd.fbody && m.lastf.fbody) goto LlastIsBetter;
+                if (!fd.fbody && !m.lastf.fbody) goto LlastIsBetter;
```

Comment:
Can't you just have one `if (!fd.fbody) goto LlastIsBetter;` ?

Reply:
Yeah but maybe the current way it's written is more clear, so I won't gonna change it. Is it okay as justification ?

Reply:
When I read it I think "why is there essentially a tautology `(!x || x)` there, is the goto label slightly different? Is fbody a function with a side-effect so it's not a tautology after all?". So I don't agree it's clearer. OTOH I don't want to bikeshed, and ultimately it's up to someone with merge rights to approve it.

Reply:
the semantic check is clear when the the relational expression is an andand.

Reply:
What happens when both have a body? If we apply your reasoning to the end, we should express that case also (with an empty if body).

Reply:
ok done.